### PR TITLE
Fix empty column default type

### PR DESCRIPTION
**Pull request by vasil-pashov**
#### Reference Issues/PRs
Closes: #1377
#### What does this implement or fix?
* Check if there are any elements inside a column and if there are no elements set the column type as empty type.
* Add tests.
#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [ ] Have you updated the relevant docstrings, documentation and copyright notice?
 - [ ] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [ ] Are API changes highlighted in the PR description?
 - [ ] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py [hunk 1]
```diff
@@ -791,26 +791,6 @@ def test_empty_pd_series(basic_store):
     assert basic_store.read(sym).data.empty
 
 
-def test_empty_pd_series_type_preservation(basic_store):
```
Comment (vasil-pashov) [inline python/tests/integration/arcticdb/version_store/test_basic_version_store.py:794]: ArcticDB should not preserve the type anymore. Empty columns will be assigned the (internal) empty type, which translates to Python's object type.

File: python/tests/nonreg/arcticdb/version_store/test_nonreg_specific.py [hunk 2]
```diff
@@ -274,36 +274,6 @@ def test_update_with_empty_dataframe_with_index(lmdb_version_store):
...
-    [
-        (np.uint8, np.uint8, TypeDescriptor.ValueType.UINT, TypeDescriptor.SizeBits.S8),
-        (np.int32, np.int32, TypeDescriptor.ValueType.INT, TypeDescriptor.SizeBits.S32),
-        ("datetime64[ns]", "datetime64[ns]", TypeDescriptor.ValueType.NANOSECONDS_UTC, TypeDescriptor.SizeBits.S64),
-
-        # For rationale see: https://github.com/man-group/ArcticDB/pull/1049
-        (float, float, TypeDescriptor.ValueType.FLOAT if IS_PANDAS_TWO else TypeDescriptor.ValueType.EMPTY, TypeDescriptor.SizeBits.S64),  # noqa: E501
-        (object, object if IS_PANDAS_TWO else float, TypeDescriptor.ValueType.EMPTY, TypeDescriptor.SizeBits.S64),
-])
-def test_empty_column_handling(lmdb_version_store, input_empty_col_dtype, output_empty_col_dtype, value_type, size_bits):
```
Comment (vasil-pashov) [inline python/tests/nonreg/arcticdb/version_store/test_nonreg_specific.py:288]: Same as https://github.com/man-group/ArcticDB/pull/1378/files#r1506124795

File: python/tests/unit/arcticdb/version_store/test_empty_column_type.py [hunk 3]
```diff
@@ -5,63 +5,109 @@
...
+def dtype(request):
+    yield request.param
+
+
+@pytest.fixture(params=[pd.RangeIndex(0,0), pd.DatetimeIndex([])])
+def empty_index(request):
+    yield request.param
+
+
+class TestCanAppendToColunWithNones:
```
Comment (alexowens90) [inline python/tests/unit/arcticdb/version_store/test_empty_column_type.py:77]: Typo

File: python/tests/unit/arcticdb/version_store/test_empty_column_type.py [hunk 4]
```diff
@@ -510,6 +463,327 @@ def test_date(self, lmdb_version_store_static_and_dynamic, date_dtype):
...
+            "sym",
+            pd.DataFrame({"col": [1, 2, 3, 4]}, dtype=float_dtype, index=self.index())
+        )
+        lmdb_version_store_static_and_dynamic.update(
+            "sym",
+            pd.DataFrame({"col": [None, None]}, index=self.update_index())
+        )
+        assert_frame_equal(
+            lmdb_version_store_static_and_dynamic.read("sym").data,
+            pd.DataFrame({"col": [1, float("NaN"), float("NaN"), 4]}, index=self.index(), dtype=float_dtype)
```
Comment (alexowens90) [inline python/tests/unit/arcticdb/version_store/test_empty_column_type.py:505]: `np.nan`?

**Review by alexowens90 — APPROVED** [review]

**Pull request merged**